File: LineArt/line.py
import pygame,math

pygame.init()

W, H = 800, 600

screen =  pygame.display.set_mode((W,H))

clock = pygame.time.Clock()

class rotateCircle():
    def __init__(self, angle):
        self.angle = angle

    def draw(self):
        angle = self.angle / 360 * 2 * math.pi;
        
        for i in range(5):
            color = 255 * (i+1) / 5
            angle += 2 / 360 * 2 * math.pi;
            x = W * (0.5 + 0.33 * math.cos(angle));
            y = H * (0.33 + 0.13 * math.sin(angle));
            r = W * (0.01 + 0.003 * math.sin(angle));
            pygame.draw.circle(screen,(color,color,color),(x, y), r)

    def update(self):
        self.angle += 5

class rotateLine():
    def __init__(self, angle):
        self.angle = angle

    def draw(self):
        angle = self.angle / 360 * 2 * math.pi;

        x1 = W / 2;
        x3 = W / 2;
        y1 = H * 0.07;
        y3 = H * 0.72;

        for i in range(10):
            color = 255 * (i+1) / 10
            angle -= 1 / 360 * 2 * math.pi;
            width = 1 if angle > math.pi else 2

            x2 = W * (0.5 + math.cos(angle) * 0.3);
            y2 = H * (0.6 + math.sin(angle) * 0.05);

            pygame.draw.line(screen,(color,color,color),(x1,y1),(x2,y2),width)
            pygame.draw.line(screen,(color,color,color),(x2,y2),(x3,y3),width)

    def update(self):
        self.angle -= 5
        if self.angle < 0:
            self.angle += 360

class dropGround():
    def __init__(self, position):
        self.position = position

    def draw(self):

        for i in range(9,-1,-1):
            pos = self.position + i

            x = W / 2;
            y = H * (0.75 + 0.25 * pos * pos / 60 / 60);
            r = (pos * pos) / 3600 * (W / 2);
    
            color = 255 * (i + 1) / 10
        
            pygame.draw.ellipse(screen,(color,color,color),(x-r, y-r*0.3, 2*r,2*r*0.3))

    def update(self):
        self.position += 1
        if self.position > 60:
            self.position = 0

circles = []
for i in range(3):
    circles.append(rotateCircle(360 / 3 * i))

lines = []
for i in range(6):
    lines.append(rotateLine(360 / 6 * i))

grounds = []
for i in range(5,-1,-1):
    grounds.append(dropGround(i / 6 * 60))

running = True
while running:
    for e in pygame.event.get():
        if e.type == pygame.QUIT:
            running = False
    
    screen.fill((0,0,0))

    for c in circles:
        c.draw()
        c.update()

    for l in lines:
        l.draw()
        l.update()

    for g in sorted(grounds, key=lambda grounds: grounds.position, reverse=True):
        g.draw()
        g.update()

    pygame.display.flip()
    clock.tick(30)

pygame.quit()

File: LineArt/line.js
const canvas = document.getElementById('canvas');
const ctx = canvas.getContext('2d');
let W = window.innerWidth;
let H = window.innerHeight;

// ウィンドウサイズに合わせてキャンバスをリサイズ
function resizeCanvas() {
    W = window.innerWidth;
    H = window.innerHeight;
    canvas.width = W;
    canvas.height = H;
}



class RotateCircle {
    constructor(angle) {
        this.angle = angle;

        this.centerX = 0.5;
        this.centerY = 0.33;
        this.rangeX = 0.33;
        this.rangeY = 0.1;
        this.centerRadius = 0.01;
        this.rangeRadius = 0.003;

        this.updateAngle = 10
        this.gradationNumber = 5
        this.gradationAngle = 2
        this.degrees = 360
    }

    draw() {
        let angle = (this.angle / this.degrees) * 2 * Math.PI;

        for (let i = 0; i < this.gradationNumber; i++) {
            const color = 255 * (i + 1) / this.gradationNumber;
            angle += (this.gradationAngle / this.degrees) * 2 * Math.PI;
            const x = W * (this.centerX + this.rangeX * Math.cos(angle));
            const y = H * (this.centerY + this.rangeY * Math.sin(angle));
            const r = Math.max(W, H) * (this.centerRadius + this.rangeRadius * Math.sin(angle));

            ctx.fillStyle = `rgb(${color}, ${color}, ${color})`;
            ctx.beginPath();
            ctx.arc(x, y, r, 0, 2 * Math.PI);
            ctx.fill();
        }
    }

    update() {
        this.angle += this.updateAngle;
        if (this.angle > this.degrees){
            this.angle -= this.degrees;
        }   
    }
}

class RotateLine {
    constructor(angle) {
        this.angle = angle;
    }

    draw() {
        let angle = (this.angle / 360) * 2 * Math.PI;

        const x1 = W / 2;
        const y1 = H * 0.07;
        const x3 = W / 2;
        const y3 = H * 0.72;

        for (let i = 0; i < 10; i++) {
            const color = 255 * (i + 1) / 10;
            angle -= (1 / 360) * 2 * Math.PI;
            const width = angle < Math.PI ? 2 : 1;

            const x2 = W * (0.5 + Math.cos(angle) * 0.3);
            const y2 = H * (0.6 + Math.sin(angle) * 0.05);

            ctx.strokeStyle = `rgb(${color}, ${color}, ${color})`;
            ctx.lineWidth = width;
            ctx.beginPath();
            ctx.moveTo(x1, y1);
            ctx.lineTo(x2, y2);
            ctx.lineTo(x3, y3);
            ctx.stroke();
        }
    }

    update() {
        this.angle -= 5;
        if (this.angle < 0) this.angle += 360;
    }
}

class DropGround {
    constructor(position) {
        this.position = position;
    }

    draw() {
        for (let i = 9; i >= 0; i--) {
            const pos = this.position + i;
            const x = W / 2;
            const y = H * (0.75 + 0.25 * (pos ** 2) / 60 / 60);
            const r = (pos ** 2) / 3600 * (W / 2);

            const color = 255 * (i + 1) / 10;

            ctx.fillStyle = `rgb(${color}, ${color}, ${color})`;
            ctx.beginPath();
            ctx.ellipse(x, y, r, r * (H / W) * 0.3, 0, 0, 2 * Math.PI);
            ctx.fill();
        }
    }

    update() {
        this.position += 1;
        if (this.position > 60) this.position = 0;
    }
}

const circles = [];
for (let i = 0; i < 3; i++) {
    circles.push(new RotateCircle((360 / 3) * i));
}

const lines = [];
for (let i = 0; i < 6; i++) {
    lines.push(new RotateLine((360 / 6) * i));
}

const grounds = [];
for (let i = 5; i >= 0; i--) {
    grounds.push(new DropGround((i / 6) * 60));
}



window.addEventListener('resize', resizeCanvas);
resizeCanvas();


// アニメーション関数
function animate() {

    // 画面を黒に塗りつぶす
    ctx.fillStyle = 'black';
    ctx.fillRect(0, 0, W, H);

    for (const circle of circles) {
        circle.draw();
        circle.update();
    }

    for (const line of lines) {
        line.draw();
        line.update();
    }

    // positionの降順でソートして描画
    grounds.sort((a, b) => b.position - a.position);
    for (const ground of grounds) {
        ground.draw();
        ground.update();
    }

    setTimeout(animate, 33);

    // 次のフレームをリクエスト
//    requestAnimationFrame(animate);
}

// アニメーション開始
animate();